synapse/rest/media/v1/thumbnail_resource.py

```diff
@@ -316,9 +319,53 @@ async def _select_and_respond_with_thumbnail(
                 respond_404(request)
                 return
 
+            responder = await self.media_storage.fetch_media(file_info)
+            if responder:
+                await respond_with_responder(
+                    request,
+                    responder,
+                    file_info.thumbnail_type,
+                    file_info.thumbnail_length,
+                )
+                return
+
+            # If we can't find the thumbnail we regenerate it. This can happen
+            # if e.g. we've deleted the thumbnails but still have the original
+            # image somewhere.
```

Comment:
I think I'm missing how we know we have the original file still? Or do we expect the generate methods to explode in that case?

Reply:
That is not an unreasonable question. Yes, it will explode, but we should handle it more explicitly.